python/llm/src/ipex_llm/transformers/model.py

```diff
@@ -346,6 +352,26 @@ def from_pretrained(cls,
             kwargs["embedding_qtype"] = embedding_qtype
             model = cls.load_convert(q_k, optimize_model, *args, **kwargs)
 
+            if pipeline_parallel:
+                model_layers = ['model.embed_tokens']
+                for i in range(model.config.num_hidden_layers):
+                    model_layers.append(f'model.layers.{i}')
+                model_layers = model_layers + ['model.norm', 'lm_head']
+
+                device_map = {}
+                split_len = len(model_layers) // parallel_gpu_num
+                for i in range(parallel_gpu_num):
+                    device_map.update({key: f'xpu:{i}' for key in
+                                       model_layers[split_len * i: split_len * (i + 1)]})
+                    if i == parallel_gpu_num - 1:
+                        device_map.update({key: f'xpu:{i}' for key in
+                                           model_layers[split_len * (i + 1):]})
+
+                from accelerate import dispatch_model
+                model = dispatch_model(
+                    model, device_map=device_map, skip_keys=["past_key_value", "past_key_values"],
+                )
+
             if speculative:
```

Comment:
Is `speculative` compatible with pipeline parallel?

Reply:
no, it's not compatible now.

Reply:
Then we need to check and report error when both are enabled.

Reply:
Do we need to add the support for enabling both speculative and pipeline parallel later?

Reply:
Maybe not critical